### ISSUE TITLE
🐛 Biome ignoring config?

### ISSUE THREAD

**Issue by Micnubinub** [biomejs/biome#6459]
### Environment information

```block
MacOS 15.5 , Bun 1.2.17, Node 24.2, Biome 2.0.4
```

### What happened?

Biome is ignoring the config and replacing single quotes with double quotes

<img width="1419" alt="Image" src="https://github.com/user-attachments/assets/0f843e75-a87e-4aeb-8101-5665b003109f" />

### Expected result

Leave single quotes as the preference is for single quotes

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by siketyan** [comment]
We were testing the new release workflow, and it failed in v2.0.1, v2.0.2, and v2.0.3. The latest version v2.0.4 should work normally. For the formatter diff, did it work in v2.0.0?

**Comment by Micnubinub** [comment]
@siketyan `2.0.4` is the one ignoring my single quotes preference. Everything works best on `2.0.0`. Should these test maybe be on a beta tag?

**Comment by Micnubinub** [comment]
@siketyan it seems at some point while formatting the line `"root": false` was added to my config... probably while running the migrate on `2.0.4`. I have reverted to `2.0.0`

**Comment by ematipico** [comment]
@Micnubinub please provide a reproduction. As for what happened, shit happens and CI is hard. You surely understand. So please file the correct information so we can triage it.

**Comment by Micnubinub** [comment]
I'll add as much detail as i can. Here is my config:
```json
{
	"$schema": "https://biomejs.dev/schemas/2.0.2/schema.json",
	"assist": { "actions": { "source": { "organizeImports": "on" } } },
	"files": {
		"includes": [
			"**",
			"!**/node_modules/**",
			"!**/.idea",
			"!**/.vscode",
			"!**/*dist",
			"!**/*gen",
			"!**/*target"
		],
		"ignoreUnknown": true,
		"maxSize": 10000000
	},
	"linter": {
		"enabled": true,
		"rules": {
			"performance": {
				"noAccumulatingSpread": {
					"level": "warn"
				},
				"noBarrelFile": {
					"level": "error"
				},
				"noDelete": {
					"level": "warn"
				},
				"noReExportAll": {
					"level": "warn"
				},
				"useTopLevelRegex": {
					"level": "warn"
				}
			},
			"security": {
				"noDangerouslySetInnerHtml": {
					"level": "warn"
				},
				"noDangerouslySetInnerHtmlWithChildren": {
					"level": "warn"
				},
				"noGlobalEval": {
					"level": "warn"
				}
			},
			"nursery": {
				"useSortedClasses": {
					"fix": "safe",
					"level": "warn"
				}
			},
			"complexity": {
				"noForEach": "off",
				"useArrowFunction": "warn",
				"useOptionalChain": {
					"fix": "safe",
					"level": "warn"
				},
				"noExcessiveCognitiveComplexity": "off",
				"useLiteralKeys": "warn",
				"recommended": true,
				"noUselessContinue": {
					"level": "warn",
					"fix": "safe"
				}
			},
			"recommended": true,
			"style": {
				"noUnusedTemplateLiteral": "off",
				"useTemplate": {
					"fix": "safe",
					"level": "warn"
				},
				"noUselessElse": {
					"fix": "safe",
					"level": "warn"
				},
				"noParameterAssign": "error",
				"useAsConstAssertion": "error",
				"useDefaultParameterLast": "error",
				"useEnumInitializers": "error",
				"useSelfClosingElements": "error",
				"useSingleVarDeclarator": "error",
				"useNumberNamespace": "error",
				"noInferrableTypes": "error",
				"noEnum": {
					"level": "warn"
				},
				"noSubstr": {
					"level": "warn"
				}
			},
			"correctness": {
				"noUnsafeOptionalChaining": {
					"level": "warn"
				},
				"noUnusedVariables": {
					"fix": "none",
					"level": "warn"
				},
				"noUnusedImports": {
					"fix": "safe",
					"level": "warn"
				},
				"noUnusedFunctionParameters": {
					"fix": "safe",
					"level": "warn"
				},
				"useIsNan": "warn",
				"noUnreachable": "warn",
				"useExhaustiveDependencies": "off"
			},
			"a11y": {
				"useKeyWithClickEvents": "off"
			},
			"suspicious": {
				"noExplicitAny": "off",
				"noApproximativeNumericConstant": {
					"level": "warn",
					"fix": "safe"
				},
				"noArrayIndexKey": "warn",
				"noAssignInExpressions": "warn",
				"noAsyncPromiseExecutor": "warn",
				"noDoubleEquals": "error",
				"noDebugger": "error",
				"noDuplicateAtImportRules": "warn",
				"noDuplicateCase": "warn",
				"noDuplicateClassMembers": "warn",
				"noDuplicateFontNames": "warn",
				"noDuplicateJsxProps": "error",
				"noDuplicateObjectKeys": "error",
				"noDuplicateParameters": "error",
				"noExtraNonNullAssertion": {
					"level": "warn",
					"fix": "safe"
				},
				"noFunctionAssign": "error",
				"noGlobalAssign": "error",
				"noDuplicateElseIf": {
					"level": "warn"
				}
			}
		}
	},
	"formatter": {
		"enabled": true,
		"formatWithErrors": true,
		"indentStyle": "tab",
		"indentWidth": 2,
		"lineWidth": 80,
		"includes": [
			"**",
			"!**/*node_modules",
			"!**/*dist",
			"!**/.idea",
			"!**/*.vscode",
			"!**/*.next",
			"!**/*.out",
			"!**/*.out-tsc"
		]
	},
	"javascript": {
		"formatter": {
			"enabled": true,
			"quoteStyle": "single",
			"jsxQuoteStyle": "single",
			"indentStyle": "tab",
			"bracketSpacing": true,
			"bracketSameLine": true,
			"indentWidth": 2,
			"trailingCommas": "none",
			"semicolons": "always",
			"lineWidth": 80
		}
	}
}
```
1. Go from any version below `2.0.2` and update to `2.0.2`
2. You get this error:
```js
~/node_modules/@biomejs/biome/bin/biome:65
                throw result.error;
                ^

Error: spawnSync ~/node_modules/@biomejs/cli-darwin-arm64/biome EACCES
    at Object.spawnSync (node:internal/child_process:1120:20)
    at Object.spawnSync (node:child_process:878:24)
    at Object.<anonymous> (~/node_modules/@biomejs/biome/bin/biome:47:42)
    at Module._compile (node:internal/modules/cjs/loader:1734:14)
    at Object..js (node:internal/modules/cjs/loader:1899:10)
    at Module.load (node:internal/modules/cjs/loader:1469:32)
    at Module._load (node:internal/modules/cjs/loader:1286:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
    at Module.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:152:5) {
  errno: -13,
  code: 'EACCES',
  syscall: 'spawnSync ~/node_modules/@biomejs/cli-darwin-arm64/biome',
  path: '~/node_modules/@biomejs/cli-darwin-arm64/biome',
  spawnargs: [ 'migrate', '--write' ]
}
```
3. Quick fix: `chmod +x node_modules/@biomejs/cli-darwin-arm64/biome`
4. Migrate: `bun run biome migrate --write`
5. Update to `2.0.4`
6. Migrate
7. At some point in there `"root": false` is added to my config

Is there any thing in your code that adds that line? 
Removing it fixed the formatting issue

**Comment by Micnubinub** [comment]
@ematipico @siketyan I've tried to downgrade and upgrade again but can't get the `"root": false`, but can't seem to get it to add it again, I hope it's not some weird edge case that's impossible to find

**Comment by ematipico** [comment]
@Micnubinub 

Please, again, provide a reproduction. The automated message should provide all info to do so. Place a readme that tells us what to do, what's the error and the expected result

**Comment by hyoretsu** [comment]
Here it is: https://github.com/hyoretsu/biome-repro-1750801289066

Extension does something different from CLI. It still happens after setting `biome.configurationPath` and even after explicitly excluding package.json from `assist`

**Comment by hyoretsu** [comment]
It's different from the behavior when using "Format Document With..." and somehow happens even after disabling formatters for JSON files, which would suggest it isn't Biome's fault, but it stops happening as soon as you disable the extension. The extension itself also flags the lines as being errored.
```
"[json][jsonc]": {
	"editor.codeActionsOnSave": {},
	"editor.defaultFormatter": null,
},
```

**Comment by hyoretsu** [comment]
Sorting imports is also weird in extension, it places "@nestjs/common" above "node:crypto" (maybe it reads node:crypto simply as crypto? CLI places node:crypto above @nestjs)
